src/test/java/jenkins/plugins/openstack/compute/JCloudsSlaveTest.java

```diff
@@ -0,0 +1,197 @@
+package jenkins.plugins.openstack.compute;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.sameInstance;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.jenkinsci.plugins.cloudstats.ProvisioningActivity;
+import org.junit.After;
+import org.junit.Test;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.openstack4j.model.compute.Address;
+import org.openstack4j.model.compute.Addresses;
+import org.openstack4j.model.compute.Server;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import hudson.ExtensionList;
+import hudson.model.Hudson;
+import hudson.model.LabelFinder;
+import hudson.model.Node;
+import hudson.model.labels.LabelAtom;
+import hudson.slaves.EnvironmentVariablesNodeProperty;
+import hudson.slaves.NodeProperty;
+import jenkins.model.Jenkins;
+import jenkins.model.Nodes;
+import jenkins.model.TestJenkins;
+
+public class JCloudsSlaveTest {
```

Comment:
There wasn't a unit-test class of this name before, and I've only implemented tests for the functionality I added, which is why the tests here _only_ check how the constructor handles various combinations of NodeProperties.